src/ducks/classifications.js

```diff
@@ -121,10 +118,128 @@ const createClassification = () => {
   };
 };
 
+/*  saveAllQueuedClassifications() attempts to individually submit
+    Classifications to Panoptes. What happens is that:
+    - When a new Classification is meant to be submitted, it is placed in a
+      queue via queueClassification()
+    - Then, each item in the queue is individually processed.
+      (The queue is emptied at this point, and a new, empty queue is prepared.)
+      - If the item (Classification) is successfully submitted/saved, that's it.
+        (It's effectively removed from the queue.)
+      - If the item fails, it's reinserted into the new queue, to be processed
+        the next time a new Classification is submitted.
+    - Once ALL the items are processed, we are now ready to prepare our data
+      for the next Subject to be fetched.
+ */
+const saveAllQueuedClassifications = (dispatch, user = null) => {
+  const QUEUE_NAME = (user)
+    ? user.id + '.' + CLASSIFICATIONS_QUEUE_NAME
+    : '_.' + CLASSIFICATIONS_QUEUE_NAME;
+        
+  const queue = JSON.parse(localStorage.getItem(QUEUE_NAME));
+
+  if (queue && queue.length !== 0) {
+    //Prepare 
+    const newQueue = [];
+    localStorage.setItem(QUEUE_NAME, null);  //Empty the queue first
+    
+    //Keep track of items processed so we know when ALL items are processed.
+    let itemsProcessed = 0;
+    let itemsFailed = 0;
+    const itemsToProcess = queue.length;
+    
+    queue.forEach((classificationData) => {
+      //Let's send those Classifications!
+      apiClient.type('classifications').create(classificationData).save()
+      
+      //SUCCESS
+      .then((classificationObject) => {
+        console.info('ducks/classifications.js saveAllQueuedClassifications() success: item ', classificationObject.id);
+
+        try {
+          Split.classificationCreated(classificationObject);
+        } catch (err) { console.error('Split.classificationCreated() error: ', err); }
+
+        //Record locally that the Classification has been seen.
+        const { workflow, subjects } = classificationObject.links;
+        dispatch(addAlreadySeen(workflow, subjects));
+        
+        //OK, we don't need you any more, Classification. Why is this here? It's in PFE.
+        //TODO: check
+        classificationObject.destroy();
```

Comment:
I'm thinking this is here because we create another classification object with the same data, so we then have to delete the resource that was initially kept in the queue. However, I'm asking around in Slack.

Reply:
👌

Reply:
I can confirm this isn't deleting the desired classification though. I've confirmed that by submitted a classification on staging and later querying the API for the submitted classification ID, which returns the recently submitted item.